benchmark/benchmark.c

```diff
@@ -201,6 +201,7 @@ void notified(microkit_channel ch)
 {
     switch (ch) {
     case START:
+#ifdef MICROKIT_CONFIG_benchmark
```

Comment:
Please can we get rid of all #ifdefs in C code.  Until we get the benchmarking working right in the RELEASE seL4, we should have a compile-time assertion that this file cannot be compiled except in benchmark mode, and just exclude it from the build for non-benchmark builds.

Reply:
This is probably a job for a different pull request, since in the echo server example, the benchmark PD plays a crucial role in the system even if it is not being built in benchmark config. It is the parent PD of every PD in the echo server (meaning it has TCB access which is required for benchmarking, but faults of its children are also directed to it which is used for printing out details of the fault). It also initialises the idle thread telling it to start cycle counting. Removing this PD would require enough restructuring of the echo server that it requires its own PR.